files/class-vip-filesystem.php

```diff
@@ -385,4 +410,77 @@ public function filter_get_attached_file( $file, $attachment_id ) {
 
 		return $file;
 	}
+
+	/**
+	 * Filter `cron_schedules` output
+	 *
+	 * Add a custom schedule for a 5 minute interval
+	 *
+	 * @param   array   $schedule
+	 *
+	 * @return  mixed
+	 */
+	public function filter_cron_schedules( $schedule ) {
+		$schedule['five_minutes'] = [
+			'interval' => 300,
+			'display' => __( 'Once every 5 minutes' ),
+		];
+
+		return $schedule;
+	}
+
+	public function schedule_update_job() {
+		if ( get_option( self::OPT_ALL_FILESIZE_PROCESSED ) ) {
+			wp_clear_scheduled_hook( self::CRON_EVENT_NAME );
+
+			return;
+		}
+
+		if (! wp_next_scheduled ( self::CRON_EVENT_NAME )) {
+			wp_schedule_event(time(), 'five_minutes', self::CRON_EVENT_NAME );
+		}
+
+		add_action( self::CRON_EVENT_NAME, [ $this, 'update_attachment_meta' ] );
+	}
+
+	/**
+	 * Cron job to update attachment metadata with file size
+	 */
+	public function update_attachment_meta() {
+		trigger_error(
+			sprintf( 'Starting %s... $vip-go-streams-debug', self::CRON_EVENT_NAME ),
+			E_USER_NOTICE );
+		if ( get_option( self::OPT_ALL_FILESIZE_PROCESSED ) ) {
+			// already done. Nothing to update
+			trigger_error(
+				sprintf( 'Already completed. Exiting %s... $vip-go-streams-debug', self::CRON_EVENT_NAME ),
+				E_USER_NOTICE );
+			return;
+		}
+
+		$updater = new Meta_Updater( 1000 );
+		$max_id = $updater->get_max_id();
```

Comment:
We should probably store the max ID in an option. It will keep shifting over time and could mean that on some sites we never hit the end. Plus, new media will get the filesize on upload so we don't need to worry about them.

Basically:

```suggestion
		$max_id = (int) get_option( self::OPT_MAX_POST_ID );
		if ( ! $max_id ) {
			$max_id = $updater->get_max_id();
			update_option( self::OPT_MAX_POST_ID, $max_id, false );
		}
```